### PR TITLE
add CI workflow

### PR DESCRIPTION
**Pull request by Lunarmagpie**
Adds CI for ruff and black

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,18 @@ The file should be put in the corresponding folder inside `src`, e.g. a
 `__post_init__` example should be located in `src/library-lagoon/post_init.py`
 (see categories in the [spreadsheet]).
 
+Your code must pass CI. To check if your code will pass CI locally run these commands:
+```
+pip install -r requirements.txt
```
Comment (trag1c) [inline README.md:30]: shouldn't it be `requirements-dev.txt`?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,18 @@ The file should be put in the corresponding folder inside `src`, e.g. a
...
+black --check .
+ruff .
```
Comment (trag1c) [inline README.md:32]: `ruff check .` is the preferred way now, `ruff .` is only kept for backwards compatibility I believe

**Review by trag1c — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (trag1c) [reply README.md:30]: why?

Comment (Lunarmagpie) [reply README.md:30]: This project doesn't have any runtime dependencies so I don't think it makes sense to split up non-existent runtime deps and the development deps.

Comment (trag1c) [reply README.md:30]: yeah i was having similar thoughts, good point

File: README.md [hunk 3]
```diff
@@ -25,6 +25,18 @@ The file should be put in the corresponding folder inside `src`, e.g. a
...
+Your code must pass CI. To check if your code will pass CI locally run these commands:
+```
+pip install -r requirements.txt
+black --check .
+ruff check .
+```
+You can also apply fixes if these checks don't pass:
+```
+black .
+ruff . --fix
```
Comment (trag1c) [inline README.md:37]:
```suggestion
ruff check . --fix
```

**Review by trag1c — CHANGES_REQUESTED** [review]

**Review by trag1c — APPROVED** [review]

**Pull request merged**